diff --git a/py_func.py b/py_func.py
index d06db36..212a6e8 100644
--- a/py_func.py
+++ b/py_func.py
@@ -164,8 +164,8 @@ def getFeatEngyScaler(feat, engy):
 
     feat_b = feat_scaler.transform(np.zeros((1, b)))
     feat_a = feat_scaler.transform(np.ones((1, b))) - feat_b
-    engy_b = engy_scaler.transform(0)
-    engy_a = engy_scaler.transform(1) - engy_b
+    engy_b = engy_scaler.transform(np.array([0]).reshape(1, 1))
+    engy_a = engy_scaler.transform(np.array([1]).reshape(1, 1)) - engy_b
 
     return feat_a, feat_b, engy_a, engy_b
 
@@ -597,4 +597,4 @@ def NVE(params):
                 for iAtom in range(len(R1)):
                     print("Si", R0[iAtom, 0], R0[iAtom, 1], R0[iAtom, 2], V0[iAtom, 0], V0[iAtom, 1], V0[iAtom, 2],
                           Fp[iAtom, 0], Fp[iAtom, 1], Fp[iAtom, 2])
-                sys.stdout.flush()
\ No newline at end of file
+                sys.stdout.flush()
diff --git a/run/run.sh b/run/run.sh
index 0293469..06249b0 100644
--- a/run/run.sh
+++ b/run/run.sh
@@ -11,18 +11,18 @@ test_set="../Data/MOVEMENT.test"
 echo "STARTING..." `date`
 
 echo STEP 1 `date`
-python ../main.py $initial_set --task -1 --validationSet $validation_set --testSet $test_set --iGPU $iGPU --dcut 6.2 --Rcut 1.9 --n2bBasis 16 --n3bBasis 4
+#python ../main.py $initial_set --task -1 --validationSet $validation_set --testSet $test_set --iGPU $iGPU --dcut 6.2 --Rcut 1.9 --n2bBasis 16 --n3bBasis 4
 
 echo STEP 2 `date`
 python ../main.py $initial_set --task -2 --validate 100 --test 0 --chunkSize 0 --epoch 500 --iGPU $iGPU --dcut 6.2 --Rcut 1.9 --learningRate 0.0001 --n2bBasis 16 --n3bBasis 4 --nL1Nodes 40 --nL2Nodes 40
 
 echo STEP 3 `date`
-python ../main.py $training_set --task -1 --restart --iGPU $iGPU
+#python ../main.py $training_set --task -1 --restart --iGPU $iGPU
 
 echo STEP 4 `date`
-python ../main.py $training_set --task -2 --restart --validate 10 --test 10 --epoch 50 --chunkSize 10000 --iGPU $iGPU
+#python ../main.py $training_set --task -2 --restart --validate 10 --test 10 --epoch 50 --chunkSize 10000 --iGPU $iGPU
 
 echo STEP 5 `date`
-python ../main.py $training_set --task -3 --restart --validate 1 --test 1 --epoch 25 --testSet $test_set --validationSet $validation_set --iGPU $iGPU --feRatio 1.0 --Rcut 1.9
+#python ../main.py $training_set --task -3 --restart --validate 1 --test 1 --epoch 25 --testSet $test_set --validationSet $validation_set --iGPU $iGPU --feRatio 1.0 --Rcut 1.9
 
 echo "FINISHED! " `date`
